types/passport.socketio/index.d.ts

```diff
@@ -0,0 +1,52 @@
+// Type definitions for passport.socketio 3.7
+// Project: https://github.com/jfromaniello/passport.socketio#readme
+// Definitions by: AhmedMKamal <https://github.com/AhmedMKamal>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+import { Store } from 'express-session';
+import { RequestHandler } from 'express';
+import { PassportStatic } from 'passport';
+import { Server, Socket } from 'socket.io';
+
+interface PassportSocketIoOptions {
+    /**
+     * the name of the cookie where express/connect stores its session_id.
+     */
+    key?: string;
+
+    /**
+     * the session_secret to parse the cookie.
+     */
+    secret?: string;
+
+    /**
+     * we NEED to use a sessionstore. No MemoryStore please.
+     */
+    store: Store;
+
+    /**
+     * the same middleware you registrer in express.
+     */
+    cookieParser?: RequestHandler;
+    passport?: PassportStatic;
+
+    /**
+     * callback on success.
+     */
+    success?: (data: any, accept: (err?: any, accepted?: boolean) => void) => void;
+
+    /**
+     * callback on fail/error.
+     */
+    fail?: (data: any, message: string, critical: string, accept: (err?: any, accepted?: boolean) => void) => void;
+}
+
+interface PassportSocketIOStatic {
+    authorize(options: PassportSocketIoOptions): (socket: Socket, fn: (err?: any) => void) => void;
+    filterSocketsByUser(io: Server, filter: (user: any) => boolean): Socket[];
+}
+
+declare const PassportSocketIO: PassportSocketIOStatic;
+
+export = PassportSocketIO;
```

Comment:
Instead of using `export =` write this as
```ts
export function authorize(options: PassportSocketIoOptions): (socket: Socket, fn: (err?: any) => void) => void;
export function filterSocketsByUser(io: Server, filter: (user: any) => boolean): Socket[];
```

Reply:
What about the default export?

Reply:
There is no need for default export. Refer to http://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html

Reply:
`export function` or `export declare function`?

Reply:
`export declare function` sorry about that. Side effect of writing more ts files that .d.ts files.

Reply:
No worries, I understand.

Reply:
I've changed my code to what you suggested but I'm getting the following errors
```
ERROR: 12:11  strict-export-declare-modifiers  All declarations in this module are exported automatically. Prefer to explicitly write 'export' for clarity. If you have a good reason not to export this declaration, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md                                                                                                              ERROR: 45:8   strict-export-declare-modifiers  'declare' keyword is redundant here. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md   ERROR: 46:8   strict-export-declare-modifiers  'declare' keyword is redundant here. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
```

Reply:
here's the complete log

```
                                                                                                                                                                                    > definitely-typed@0.0.3 test E:\WorkSpace\open-source\DefinitelyTyped                                                                                                              > node node_modules/types-publisher/bin/tester/test.js --run-from-definitely-typed "passport.socketio"                                                                                                                                                                                                                                                                  Clean data                                                                                                                                                                          Clean logs                                                                                                                                                                          Clean output                                                                                                                                                                        Using local Definitely Typed at E:\WorkSpace\open-source\DefinitelyTyped.                                                                                                           Parsing definitions...                                                                                                                                                              Found 6086 packages.                                                                                                                                                                Parsing in parallel...                                                                                                                                                              Running: git rev-parse --verify master                                                                                                                                              bd70841d08551b27ee2304663d72fbf546099e4a                                                                                                                                            Running: git diff master --name-status                                                                                                                                              M       types/passport.socketio/index.d.ts                                                                                                                                          Testing 1 changed packages: passport.socketio                                                                                                                                       Testing 0 dependent packages:                                                                                                                                                       Running with 2 processes.                                                                                                                                                           Installing NPM dependencies...                                                                                                                                                        E:\WorkSpace\open-source\DefinitelyTyped/types/node: npm install --ignore-scripts --no-shrinkwrap --no-package-lock --                                                            no-bin-links --no-save                                                                                                                                                               from E:\WorkSpace\open-source\DefinitelyTyped/types/node: up to date in 1.283s                                                                                                     found 0 vulnerabilities                                                                                                                                                             Running: node E:\WorkSpace\open-source\DefinitelyTyped\node_modules\dtslint\bin\index.js --installAll                                                                               Cleaning old installs and installing for all TypeScript versions...                                                                                                                 Working...                                                                                                                                                                          Installing to C:\Users\enahm\.dts\typescript-installs\2.0...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\2.1...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\2.2...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\2.3...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\2.4...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\2.5...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\2.6...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\2.7...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\2.8...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\2.9...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\3.0...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\3.1...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\3.2...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\3.3...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\3.4...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\3.5...
Installed!

Installing to C:\Users\enahm\.dts\typescript-installs\next...
Installed!
Testing...
1> passport.socketio START
















1> passport.socketio failing:
1> Error: E:/WorkSpace/open-source/DefinitelyTyped/types/passport.socketio/index.d.ts:12:11
1> ERROR: 12:11  strict-export-declare-modifiers  All declarations in this module are exported automatically. Prefer to                                                             explicitly write 'export' for clarity. If you have a good reason not to export this declaration, add 'export {}' to the                                                             module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare                                                            -modifiers.md
1> ERROR: 45:8   strict-export-declare-modifiers  'declare' keyword is redundant here. See: https://github.com/Microsoft                                                            /dtslint/blob/master/docs/strict-export-declare-modifiers.md
1> ERROR: 46:8   strict-export-declare-modifiers  'declare' keyword is redundant here. See: https://github.com/Microsoft                                                            /dtslint/blob/master/docs/strict-export-declare-modifiers.md
1>
1>     at E:\WorkSpace\open-source\DefinitelyTyped\node_modules\dtslint\bin\index.js:190:19
1>     at Generator.next (<anonymous>)
1>     at fulfilled (E:\WorkSpace\open-source\DefinitelyTyped\node_modules\dtslint\bin\index.js:5:58)


=== PERFORMANCE ===

{
  "passport.socketio": 15207,
}


=== ERRORS ===



Error in passport.socketio
Error: E:/WorkSpace/open-source/DefinitelyTyped/types/passport.socketio/index.d.ts:12:11
ERROR: 12:11  strict-export-declare-modifiers  All declarations in this module are exported automatically. Prefer to exp                                                            licitly write 'export' for clarity. If you have a good reason not to export this declaration, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md                                                                                                              ERROR: 45:8   strict-export-declare-modifiers  'declare' keyword is redundant here. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md   ERROR: 46:8   strict-export-declare-modifiers  'declare' keyword is redundant here. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md                                                                                                                                                                                           at E:\WorkSpace\open-source\DefinitelyTyped\node_modules\dtslint\bin\index.js:190:19                                                                                                at Generator.next (<anonymous>)                                                                                                                                                     at fulfilled (E:\WorkSpace\open-source\DefinitelyTyped\node_modules\dtslint\bin\index.js:5:58)                                                                                  Error: The following packages had errors: passport.socketio                                                                                                                             at doRunTests (E:\WorkSpace\open-source\DefinitelyTyped\node_modules\types-publisher\src\tester\test-runner.ts:240:11)                                                              at process._tickCallback (internal/process/next_tick.js:68:7)                                                                                                                   npm ERR! code ELIFECYCLE                                                                                                                                                            npm ERR! errno 1                                                                                                                                                                    npm ERR! definitely-typed@0.0.3 test: `node node_modules/types-publisher/bin/tester/test.js --run-from-definitely-typed "passport.socketio"`                                        npm ERR! Exit status 1                                                                                                                                                              npm ERR!                                                                                                                                                                            npm ERR! Failed at the definitely-typed@0.0.3 test script.                                                                                                                          npm ERR! This is probably not a problem with npm. There is likely additional logging output above.                                                                                                                                                                                                                                                                      npm ERR! A complete log of this run can be found in:                                                                                                                                npm ERR!     C:\Users\enahm\AppData\Roaming\npm-cache\_logs\2019-08-21T20_48_15_024Z-debug.log 
```

Reply:
You need to export `interface PassportSocketIoOptions` as well

Reply:
I'm afraid it's not the reason, still getting

```
1> ERROR: 45:8  strict-export-declare-modifiers  'declare' keyword is redundant here. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md 
1> ERROR: 46:8  strict-export-declare-modifiers  'declare' keyword is redundant here. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md 
```

Reply:
remove the `declare` keyword then

Reply:
Yes I did that, and I'm waiting for the test to finish

Reply:
Okay, It was my fault to add `declare` from the beginning 😅